src/Host/Broker/Impl/UserProfile/UserProfileManager.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Microsoft.Common.Core;
+using Microsoft.Common.Core.Json;
+using Microsoft.Extensions.Logging;
+using Microsoft.R.Host.Protocol;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO.Pipes;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.R.Host.Broker.UserProfile {
+    public class UserProfileManager {
+        private readonly ILogger _logger;
+
+        public UserProfileManager(ILogger<UserProfileManager> logger) {
+            _logger = logger;
+        }
+
+        public Task<RUserProfileServiceResponse> CreateProfileAsync(RUserProfileServiceRequest request, CancellationToken ct) {
+            return ProfileWorkerAsync(NamedPipeServerStreamFactory.CreatorName, Resources.Error_ProfileCreationFailedIO, request, ct);
+        }
+
+        public Task<RUserProfileServiceResponse> DeleteProfileAsync(RUserProfileServiceRequest request, CancellationToken ct) {
+            return ProfileWorkerAsync(NamedPipeServerStreamFactory.DeletorName, Resources.Error_ProfileDeletionFailedIO, request, ct);
+        }
+
+        private async Task<RUserProfileServiceResponse> ProfileWorkerAsync(string name, string log, RUserProfileServiceRequest request, CancellationToken ct) {
+            using (NamedPipeClientStream client = new NamedPipeClientStream(name)) {
+                try {
+                    await client.ConnectAsync(ct);
+
+                    string jsonReq = JsonConvert.SerializeObject(request);
```

Comment:
It calls `Dispose` on the underlying `StreamWriter`, which will in turn call the same on the underlying `Stream`. But `StreamWriter` has a constructor that lets you tell it to leave the stream alone - try that:

https://msdn.microsoft.com/EN-US/library/gg712853(v=VS.110,d=hv.2).aspx